client/app/vps/cloud-database/order/vps-cloud-database-order.controller.js

```diff
@@ -0,0 +1,176 @@
+            });
+    }
+
+    cancelFurtherChoices () {
+        this.$timeout.cancel(this.redirectionTimeout);
+        this.loading.purchaseOrder = false;
+        this.loading.redirection = false;
+
+        this.currentOrder.duration = null;
+        this.currentOrder.contractsAccepted = false;
+        this.purchaseOrder = null;
+    }
+
+    getPricesForEachDuration (durations, version, ram, datacenter) {
+        return this.$q.all(_.map(
+            durations,
+            duration => this.getPrices(duration, version, ram, datacenter)
+        )).catch(error => this.CloudMessage.error([
+            this.$translate.instant("vps_tab_cloud_database_order_fetch_prices_failed"),
+            _(error).get("data.message", error)
+        ].join(" ")));
+    }
+
+    getPrices (duration, version, ram, datacenter) {
+        return this.ApiOrderDb.getNewDetails({ duration: duration.value, version, ram, datacenter }).$promise
+            .then(details => {
+                duration.details = details;
+            });
+    }
+
+    canOrder () {
+        return !_.any(this.loading) &&
+            this.currentOrder.version &&
+            this.currentOrder.ram &&
+            this.currentOrder.datacenter &&
+            this.currentOrder.duration &&
+            this.currentOrder.contractsAccepted;
+    }
+
+    generatePurchaseOrder () {
+        this.loading.purchaseOrder = true;
+        this.loading.redirection = true;
+        this.ApiOrderDb.orderNew(
+            { duration: this.currentOrder.duration.value },
+            {
+                version: this.currentOrder.version.value,
+                ram: this.currentOrder.ram.value,
+                datacenter: this.currentOrder.datacenter.value,
+                offer: "public"
+            }).$promise
+            .then(order => {
+                this.purchaseOrder = order;
+                this.redirectionTimeout = this.$timeout(() => {
+                    this.loading.redirection = false;
+                    this.openPurchaseOrder();
+                }, 5000);
+            })
+            .catch(error => this.CloudMessage.error([
+                this.$translate.instant("vps_tab_cloud_database_order_failed"),
+                _(error).get("data.message", error)
+            ].join(" ")))
+            .finally(() => {
+                this.loading.purchaseOrder = false;
+            });
+    }
+
+    openPurchaseOrder () {
+        this.atInternet.trackOrder({
+            name: `[sql-public]::${this.currentOrder.version.value}`,
+            page: "web::payment-pending",
+            orderId: this.purchaseOrder.orderId,
+            priceTaxFree: this.purchaseOrder.prices.withoutTax.value,
+            price: this.purchaseOrder.prices.withTax.value,
+            status: 1
+        });
+        this.$window.open(this.purchaseOrder.url);
+    }
+}
+
+angular.module("managerApp").controller("VpsCloudDatabaseOrderCtrl", VpsCloudDatabaseOrderCtrl);
```

Comment:
Nice but you expose `VpsCloudDatabaseOrderCtrl ` to all Javascript files there (`document. VpsCloudDatabaseOrderCtrl !== undefined`), you should create a new context:

```js
{
    class VpsCloudDatabaseOrderCtrl { ... } 

   angular
        .module("managerApp")
        .controller("VpsCloudDatabaseOrderCtrl", VpsCloudDatabaseOrderCtrl);
}
```

Reply:
Well, I guess all the controllers in Cloud are exposed then.

Reply:
It doesn't look like an issue to me. It's useful to have it in the debugger context.